### PR TITLE
Make slideextract self-contained and cross-platform

### PR DESCRIPTION
**Pull request by aminya**
- Now the library works on all platforms. Tested it on Windows, and it works! 🎉 
- Added Cmake and Conan for installation and building of the packages.
- Replaced Linux specific header file (unistd) with the awesome cargs library which is cross-platform and nicer to work with.
- Updated the readme
- Fixed the generation of empty images
- Support selecting the region from all directions in GUI

### CODE REVIEW

File: main.c [hunk 1]
```diff
@@ -28,14 +28,14 @@
...
-#include <unistd.h>
+#include <cargs.h>
 #include <stdbool.h>
 
 //! Print version and exit program
 static void
 version ()
 {
-	puts("slideextract " VERSION "\n\
+	puts("slideextract 0.1.0 \n\
```
Comment (aminya) [inline main.c:38]: I found manually setting of this number simplifies the code. We can make it a compile-time value if that is needed.

**Comment by aminya** [comment]
@szanni Check this out!

I am also interested to add continuous integration for providing binaries to make it easier for the average user to use this awesome tool.

**Comment by szanni** [comment]
Thanks for the effort and the hidden bug report but I hope this does not come off as too hash - pretty much the whole patch set won't fly...

- Thanks for reporting that the comparison region could only be selected from top-left to bottom right. This was a bug. But this can and should be solved by simple math and not 30 lines of added temp variables, if's, and convoluted logic. See 35cde374b4f1bc1ecd0be2597dc8e39a8390bf1b
- There is no way I will add Cmake. If you had seen the CHANGELOG (yeah who reads those right?) you would have seen that I just removed the monstrosity called autotools. No way will I add back something similar.
- The same goes for conan. Never heard of it. It does not even have a Wikipedia entry and seems to be some type of corporate backed advertising thingy. Apparently QT6 added some integration?! But a no from my side until the platform looks mature.
- `<cargs.h>` looks nice, the function naming IMO is crap. Not your fault but again nope. If it was a single header file library - then maybe - but not in this form.
- And I dislike making anything even *more* manual.

Regarding windows: does anyone use windows command line tools for stuff like this these days? I though command line inclined people were using WSL2 these days anyways?
Regarding mac: mac has macports and homebrew if anyone feels inclined to add support.

Regarding CI: the free services are changing all the time, I don't feel like constantly fixing this. A manual test suite would be amazing though. Automated CI IMO is warranted for bigger projects with many moving parts but for a small tool like this most likely more work than work saving.

And finally regarding binaries: there is no way I will distribute anything the like. If you feel so inclined - be my guest - I will not. This is due to opencv including tons of patented tech - video codecs and the like. It is beautiful we have open source projects like VLC media player, they are however backed by the VideoLan organization and a lot of people would jump in to fight a legal battle. I have no such backing and am not willing to take the risk.

**Comment by aminya** [comment]
NP

**Pull request closed without merging**